### PR TITLE
T/callback update

### PR DESCRIPTION
**Pull request by taylormjs**
## Updates to DGEB, CaLM and MoleculeACE (public) callbacks:
- Add DGEB callback to be consistent with other linear probe callbacks and to work with hydra configs
- LinearProbeCallback now properly handles mean pooling (before was including pad tokens)
- MoleculeACE Callback now matches lobster_internal
- CaLM callback now uses LinearProbeCallbacks embed method instead of its own
- DGEB callback implemented for ESM

Major Features:
- Add comprehensive DGEBEvaluationCallback for UME and ESM models
- Implement ESMAdapterDGEB for direct ESM model evaluation without checkpoints
- Add shared pooling utilities for consistent embedding aggregation
- Enhance MoleculeACE linear probe with better model compatibility

Core Components:
- DGEBEvaluationCallback: Unified callback supporting both UME (checkpoint-based) and ESM (direct) evaluation workflows
- ESMAdapterDGEB: DGEB-compatible adapter for ESM models with proper masked pooling
- Shared pooling utilities: mean/max/cls/last pooling with attention masking
- Enhanced error handling and graceful task failure recovery

Improvements:
- Better embedding extraction across different model types
- Improved linear probe callbacks with enhanced input processing
- Updated DGEB runners with better error handling and reporting
- Comprehensive test coverage for new ESM adapter functionality

## Type of Change
- [x] Bug fix
- [x] New feature  
- [ ] Documentation update
- [ ] Performance improvement
- [x] Code refactoring

### CODE REVIEW

File: src/lobster/evaluation/dgeb_mock_runner.py [hunk 1]
```diff
@@ -166,24 +166,46 @@ def run_mock_evaluation(
             "results": [],
         }
 
-        # Extract key metrics from results
+        # Extract key metrics from results with error handling for individual tasks
```
Comment (ncfrey) [inline src/lobster/evaluation/dgeb_mock_runner.py:169]: good call - are you seeing failed tasks frequently?

File: src/lobster/evaluation/esm_dgeb_adapter.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        Layers to extract embeddings from. For ESM, typically only last layer is meaningful.
+    process_and_embed_fn : Callable
+        Function to process inputs and extract embeddings from the ESM model.
+        Should have signature: (module, sequences, modality, aggregate) -> torch.Tensor
+    """
+
+    def __init__(
+        self,
+        module: L.LightningModule,
+        modality: Literal["protein", "dna"] = "protein",
```
Comment (ncfrey) [inline src/lobster/evaluation/esm_dgeb_adapter.py:50]: can we harmonize this with UME's modality types?

File: src/lobster/evaluation/esm_dgeb_adapter.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+    def _get_tokenizer(self, model_name: str):
+        """Provide a minimal tokenizer for BioSeqTransformer interface compliance.
+
+        Note: ESM sequence processing is handled entirely through process_and_embed_fn,
+        so this tokenizer is only used to satisfy the parent class interface requirements.
+        """
+
+        # Create a minimal tokenizer object with required attributes
+        class MinimalTokenizer:
```
Comment (ncfrey) [inline src/lobster/evaluation/esm_dgeb_adapter.py:122]: maybe DummyTokenizer then since this isn't used?

File: src/lobster/evaluation/dgeb_adapter.py [hunk 4]
```diff
@@ -451,24 +452,10 @@ def _extract_layer_embeddings(self, sequences: list[str], modality: LobsterModal
...
-                layer_hidden_masked = layer_hidden.masked_fill(mask == 0, float("-inf"))
-                pooled = layer_hidden_masked.max(dim=1)[0]
-            elif self.pool_type == "cls":
-                pooled = layer_hidden[:, 0, :]
-            elif self.pool_type == "last":
-                lengths = attention_mask.sum(dim=1) - 1
-                pooled = torch.stack([layer_hidden[i, l, :] for i, l in enumerate(lengths)], dim=0)
-            else:
-                raise ValueError(f"Unsupported pool_type: {self.pool_type}")
+            pooled = apply_dgeb_pooling(layer_hidden, attention_mask, self.pool_type)
```
[the remark this answers is not in the document]
Comment (taylormjs) [reply src/lobster/evaluation/dgeb_adapter.py:458]: Yeah, it's nice for easy to transfer to esm_dgeb_adapter, for example

File: src/lobster/evaluation/dgeb_mock_runner.py (hunk 1 above)
Comment (taylormjs) [reply src/lobster/evaluation/dgeb_mock_runner.py:169]: Not frequently, but enough to where I wanted a report of which tasks failed

File: src/lobster/evaluation/esm_dgeb_adapter.py (hunk 3 above)
Comment (taylormjs) [reply src/lobster/evaluation/esm_dgeb_adapter.py:122]: Yeah, good point. I had it as DummyTokenizer before but changed it because of Monday "vibes"
